### PR TITLE
CA-390883: Move usb_reset.py to python3, test mount() in a namespace

### PR DESCRIPTION
**Pull request by bernhardkaindl**
This is the requested follow-up from #5401 (CA-390883: Add unit test for usb_reset.py)

The only changes in this PR compared to #5401 which was closed due to inactivity are:
- Move `scripts/usb_reset.py` to `python3/libexec/usb_reset.py`
- Removed the Python2 compatibility (as requested by @psafont)
- Removed the needed changes to `.github/workflows/main.yml` that were merged using another PR.

#5401 was approved by linliu, and the only remaining request was by @psafont:
> It'd be nice to get this merged into the feature branch, can the testing for python 2 be dropped?

Done now.

### CODE REVIEW

**Comment by psafont** [comment]
Thanks for opening this, I've changed the base branch to feature/py3

File: python3/tests/import_helper.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
+"""helpers for unit-testing functions in scripts without permanent global mocks"""
+import os
+import sys
+from contextlib import contextmanager
+from types import ModuleType
+
+from mock import Mock
+from typing import Generator
```
Comment (psafont) [inline python3/tests/import_helper.py:8]:
```suggestion
from typing import Generator
from mock import Mock
```

**Review by psafont — APPROVED** [review]

File: python3/tests/import_helper.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    ```
+    """
+    for module_name in module_names:
+        sys.modules[module_name] = Mock()
+    yield
+    for module_name in module_names:
+        sys.modules.pop(module_name)
+
+
+def import_file_as_module(relative_script_path):  # type:(str) -> ModuleType
```
Comment (liulinC) [inline python3/tests/import_helper.py:37]: this file is duplicated with https://github.com/xapi-project/xen-api/blob/feature/py3/python3/unittest/import_file.py

**Comment by liulinC** [comment]
currently python3 folder has sub-folders as
- tests
- unittest
@stephenchengCloud  will combine these into one when it is the time for the feature merge.
Given this PR target to `feature/py3` now, I suggest the test goes to `unittest` and re-use `import-file`

File: python3/tests/import_helper.py (hunk 2 above)
Comment (psafont) [reply python3/tests/import_helper.py:37]: Not quite, this one provides more functionality

Comment (psafont) [reply python3/tests/import_helper.py:37]: I've prepared a branch that deletes import_file.py: https://github.com/xapi-project/xen-api/compare/feature/py3...psafont:xen-api:test-and-fix-usb_reset-mount

**Review by liulinC — APPROVED** [review]
`tests` and `unittest` folder can be combined, 

I approve this PR for now, @stephenchengCloud  Could you please help on this at proper time.

**Comment by stephenchengCloud** [comment]
>  @stephenchengCloud Could you please help on this at proper time.

Sure, I'll raise another PR for this.

**Pull request merged**